configure.ac

```diff
@@ -3393,15 +3433,18 @@ AC_ARG_ENABLE([sha],
     [ ENABLED_SHA=yes ]
     )
 
-if test "$ENABLED_SHA" = "no"
+if  test "$ENABLED_WOLFENGINE" != "yes" && test "$ENABLED_WOLFENGINEWITHOUTFIPS" != "yes"
```

Comment:
Hmm, what's the idea with the logic here? I'm not grasping it.

Reply:
Yeah....this particular section of the diff is rather confusing. Here is what it was before: 

```
if test "$ENABLED_SHA" = "no"
then
    AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
else
    # turn off SHA if leanpsk or leantls on
    if test "$ENABLED_LEANPSK" = "yes" || test "$ENABLED_LEANTLS" = "yes"
    then
        AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
        ENABLED_SHA=no
    fi
fi
```
After my change that looks like: 
```
if  test "$ENABLED_WOLFENGINE" != "yes" && test "$ENABLED_WOLFENGINEWITHOUTFIPS" != "yes"
then
    if test "$ENABLED_SHA" = "no"
    then
        AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
    else
        # turn off SHA if leanpsk or leantls on
        if test "$ENABLED_LEANPSK" = "yes" || test "$ENABLED_LEANTLS" = "yes"
        then
            AM_CFLAGS="$AM_CFLAGS -DNO_SHA -DNO_OLD_TLS"
            ENABLED_SHA=no
        fi
    fi
fi

```
Its the same logic but its prevented from running if one of them is enabled.